### ISSUE TITLE
Data size must be limited

### ISSUE THREAD

**Issue by yusufpapurcu** [yakuter/chanman#3]
We are not checking data size for now.
This checking needed for limiting data size because we keep data on memory and we don't want to using memory aggressively.
We should define size limit like:
```go
	opts := &chanman.Options{
		CallbackFn: callbackFn,
		Limit:      19,
		Worker:     5,
		DataSize: 1024, // This will be added
	}
```

For unlimited size we can use ```DataSize: 0```
Can someone handle this?

**Comment by halilkocaoz** [comment]
Hey @yusufpapurcu, I can handle this.

**Comment by yusufpapurcu** [comment]
Great thank you @halilkocaoz 💯

**Comment by halilkocaoz** [comment]
Implementing this limit is tough than I thought 😸 because I must deal with the size of any data in runtime. I think that using the **reflection** packet for this is a good choice.

Could you please correct me, If I'm wrong?

Tomorrow morning, I will start to re-analyze it and If I'm on the right track, it might take me a few days to implement it.

**Comment by yakuter** [comment]
Hello @halilkocaoz. This should be fine:

`size := reflect.TypeOf(T).Size()`

Thank you.

**Comment by halilkocaoz** [comment]
> Hello @halilkocaoz. This should be fine:
> 
> `size := reflect.TypeOf(T).Size()`
> 
> Thank you.

Is `Size` func enough for this?

Last night, I thought I had to write a packet for this and a few minutes ago I found an already written packet that what I want: https://github.com/DmitriyVTitov/size

**Comment by yusufpapurcu** [comment]
I think `reflect.TypeOf(T).Size()` or `unsafe.Sizeof()` are ok for this situation. They are returns actual size in memory.
There is a great [explanation](https://stackoverflow.com/a/44258164/13706529) here. First example explains.

**Comment by yakuter** [comment]
For our situation `reflect.TypeOf(T).Size()` is enough. You should avoid using unsafe unless you have no other choice.

**Comment by yakuter** [comment]
Closed with #4
